sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -697,6 +813,17 @@ public Write withTableId(String tableId) {
       return withTableId(StaticValueProvider.of(tableId));
     }
 
+    /**
+     * Returns a new {@link BigtableIO.Write} with the provided credentials.
+     *
+     * <p>Does not modify this object.
+     */
+    public Write withCredentialsProvider(CredentialsProvider credentialsProvider) {
```

Comment:
It looks like beam has its own abstraction for Crendential provider...take a look at org.apache.beam.sdk.extensions.gcp.auth.CredentialFactory. Take a look at GcpOptions and how other services use getGcpCredentials there